storage-node/src/disk/mod.rs

```diff
@@ -0,0 +1,4 @@
+/// [`DiskManager`] contains the common logic to read from a disk.
+///
+/// FIXME: Do we really need `DiskManager`? What state should it hold?
```

Comment:
Maybe disk size? I think we need a common logic to read and write to disk within certain disk size. I can look at it later!

Reply:
I believe it's the cache's responsibility to record this information to ensure it will never write out bytes exceeding its size? Besides, implementation-wise this sounds like a disk simulator but I don't think we need one in a database? Feel like I need more rationale to do this... 🥹

Reply:
Yeah, you are right, we should record the size in the cache. But I think we need a place to provide common helper function to read and write disk... Maybe we can rename it?

Reply:
Yes. And if we're just providing the common logic and not holding any state, we can use functions directly and there's no need for a struct.

Reply:
lol I just looked at the bustub. Do you think these members also make sense for us? https://github.com/cmu-db/bustub/blob/325775aadf914cf66063df7e5688e0ec5b15ded8/src/include/storage/disk/disk_manager.h#L4

Reply:
It makes some sense to have a file lock on each cached item to prevent r/w races but I'm feeling like it's the cache's responsibility to ensure the order of the operations to the files (or they might all wait for the lock on the same file and there's no guarantee who's gonna get the lock first). Also we may want to record some statistics (the number of writes, the number of bytes we write, etc.) for benchmark usage but that should be very far away in the future. But yes we can keep this for now.

Reply:
Oh I think we need at least some latches in DiskManager maybe, especially when multiple cache instances and also maybe local_fs try to r/w cache at the same time. But it is too complex to consider now (there seem a lot of things we can do to optimize under multi-threads and parallelism)
Sure let's keep it for now!

Reply:
`local_fs` is the underlying storage and it's not supposed to write the cache. It should always be the `StorageManager` that operates the cache. And I believe the cache should have a separate disk area from the `local_fs` storage.

Reply:
Oh yeah you are right about `local_fs`. Sorry for this! We can discuss the design of parallelism later.